tools/genomespace/genomespace_push.xml

```diff
@@ -12,16 +12,18 @@
         #set $input_file = $URL.split("^")[0] if "^" in $URL else $URL
 
         #assert $input_file, Exception( 'You must select a valid input file.' )
-        --json_parameter_file '${output_file1}'
-        --galaxy_root $__root_dir__
-        --data_conf $__datatypes_config__
+
+        #import binascii
+        --input_urls '${ binascii.hexlify(str($input_file).encode("utf8")) }'
     </command>
     <!-- If using this tool through bioblend, the URL parameter should contain a comma separated list of GenomeSpace URLs -->
     <inputs check_values="False">
-        <param name="URL" type="genomespacefile" label="Choose Input File from GenomeSpace" select_type="FILE" />
+        <param name="URL" type="genomespacefile" label="Choose Input File from GenomeSpace" action="IMPORT" select_type="FILE_OR_FOLDER" multiple="True" />
     </inputs>
     <outputs>
-        <data format="auto" name="output_file1" />
+        <data auto_format="true" name="output_file1">
+            <discover_datasets pattern="__name_and_ext__" directory="." visible="true" assign_primary_output="true" format="auto" recurse="true" />
```

Comment:
So this means only the first output will be usable in workflows. Maybe you can create a dataset collection when there is more than 1 file being added ? Also the ext will take preference over format auto, so I think you can drop format="auto". Finally I kind of liked the sniffing you were doing previously, this currently struggles hard for fastq or fastq.gz files :/. I think galaxy could do better here, especially for `fastq.gz`, which will simply be set to `gz`.

Reply:
Oops, I had overlooked that it wouldn't work with a workflow if it's just a single output. I'll try switching it over to a dataset collection.

If we do manual sniffing though, then the recurse flag will not really be useful, because we'd have to generate the galaxy.json file anyway. That in itself is ok, but makes the wrapper code slightly more complicated because that recursion logic would go in there too. Still, I'm not sure why the manual-sniffing is working better for fastq.gz, because we are still calling sniff.handle_uploaded_dataset_file() in: https://github.com/galaxyproject/galaxy/pull/5242/files#diff-a9715b311879402b8b86a4013e4ec56bL123 and galaxy should internally be doing the same?

Reply:
I'd have to play around with this as well, so I can't answer this right away. What I noticed is that the sniffing is less than optimal:
![screen shot 2018-01-17 at 15 51 33](https://user-images.githubusercontent.com/6804901/35048812-6c9f83c4-fb9e-11e7-8069-244e9e7477f1.png)